정우/week3/더 맵게.js

```diff
@@ -0,0 +1,84 @@
+class MinHeap {
+  constructor() {
+    this.heap = [];
+  }
+
+  size() {
+    return this.heap.length;
+  }
+
+  push(value) {
+    this.heap.push(value);
+    let currentIndex = this.heap.length - 1;
+
+    while (
+      currentIndex > 0 &&
+      this.heap[currentIndex] < this.heap[Math.floor((currentIndex - 1) / 2)]
+    ) {
+      const temp = this.heap[currentIndex];
+      this.heap[currentIndex] = this.heap[Math.floor((currentIndex - 1) / 2)];
+      this.heap[Math.floor((currentIndex - 1) / 2)] = temp;
+      currentIndex = Math.floor((currentIndex - 1) / 2);
+    }
+  }
+
+  pop() {
+    if (this.heap.length === 0) return null;
+    if (this.heap.length === 1) return this.heap.pop();
+
+    const minValue = this.heap[0];
+    this.heap[0] = this.heap.pop();
+    let currentIndex = 0;
+
+    while (currentIndex * 2 + 1 < this.heap.length) {
+      let minChildIndex =
+        currentIndex * 2 + 2 < this.heap.length &&
+        this.heap[currentIndex * 2 + 2] < this.heap[currentIndex * 2 + 1]
+          ? currentIndex * 2 + 2
+          : currentIndex * 2 + 1;
+
+      if (this.heap[currentIndex] < this.heap[minChildIndex]) {
+        break;
+      }
+
+      const temp = this.heap[currentIndex];
+      this.heap[currentIndex] = this.heap[minChildIndex];
+      this.heap[minChildIndex] = temp;
+      currentIndex = minChildIndex;
+    }
+
+    return minValue;
+  }
+
+  peek() {
+    return this.heap[0];
+  }
+}
+
+function solution(scoville, K) {
+  let answer = 0;
+  const minHeap = new MinHeap();
+
+  scoville.forEach((item) => {
+    minHeap.push(item);
+  });
+
+  while (minHeap.size() > 1) {
+    let min1 = minHeap.pop();
+    if (min1 >= K) {
+      return answer;
+    } else {
```

Comment:
이건 그냥 제 습관인데, 전 if문에 return문이 있다면 else문은 따로 기재를 안하곤합니닷 
```suggestion
    if( min1>=K) return answer;
    let min2= minHeap.pop();
    ...
```
이런식으로 ㅎㅎ 그냥 이렇게 쓰는 사람도 있다 알려주고싶었어용ㅎ

Reply:
오호 유진이가 적어준거처럼 if 문에 return 있으면 else 안해줘도 되는거 첨 알았다,,!